### PR TITLE
Many many fixes

### PR DESCRIPTION
**Pull request by marco-neumann-by**
See individual commits. That all came up when battle-testing `rle-array` on some internal project.

### CODE REVIEW

File: rle_array/array.py [hunk 1]
```diff
@@ -387,18 +442,23 @@ def __array__(self, dtype: Any = None) -> Any:
 
         return decompress(self._data, self._positions, dtype)
 
-    def astype(self, dtype: Any, copy: bool = True) -> Any:
+    def astype(self, dtype: Any, copy: bool = True, casting: str = "unsafe") -> Any:
```
Comment (hendrik-teuber-by) [inline rle_array/array.py:445]: Why is unsafe the default?

File: rle_array/array.py [hunk 2]
```diff
@@ -243,7 +264,20 @@ def _from_sequence(
...
+            if (dtype is not None) and (dtype != scalars.dtype):
+                # some cast required
+                if dtype == np.bool_:
+                    # bool case
+                    scalars, mask = coerce_to_boolean_array(scalars)
+                    if mask.any():
+                        raise TypeError("Masked booleans are not supported")
+                else:
+                    # catch-them-all case
+                    # TODO: get rid of this unsafe cast
```
Comment (hendrik-teuber-by) [inline rle_array/array.py:277]: Maybe create an Issue for this TODO

**Review by hendrik-teuber-by — APPROVED** [review]

File: rle_array/array.py (hunk 1 above)
Comment (marco-neumann-by) [reply rle_array/array.py:445]: because this is what the pandas extension array is using (aka our baseclass).

File: rle_array/array.py (hunk 2 above)
Comment (marco-neumann-by) [reply rle_array/array.py:277]: Done, #32

**Pull request merged**